CrewChiefV4/DriverTrainingService.cs

```diff
@@ -0,0 +1,336 @@
+                catch (Exception e)
+                {
+                    Console.WriteLine("Unable to complete recording session : " + e.Message);
+                }
+                isRecordingSession = false;
+            }
+        }
+
+        public static void stopRecordingMessage()
+        {
+            if (DriverTrainingService.isRecordingSound)
+            {
+                try
+                {
+                    DriverTrainingService.waveSource.StopRecording();
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine("Failed to record a training session sound " + e.Message);
+                }
+                DriverTrainingService.isRecordingSound = false;
+            }
+        }
+
+        public static void startRecordingMessage(int distanceRoundTrack)
+        {
+            if (isRecordingSession)
+            {
+                if (DriverTrainingService.isRecordingSound)
+                {
+                    Console.WriteLine("sound already being recorded");
+                }
+                else
+                {                    
+                    String fileName = distanceRoundTrack + ".wav";
+                    String recordingName = DriverTrainingService.trackName + "_" + DriverTrainingService.carClass.ToString() + "_" + fileName;                   
+                    try
+                    {
+                        lock (_lock) 
+                        { 
+                            DriverTrainingService.waveSource = new WaveInEvent();
+                            DriverTrainingService.waveSource.WaveFormat = new WaveFormat(22050, 1);
+                            DriverTrainingService.waveSource.DataAvailable += new EventHandler<WaveInEventArgs>(waveSource_DataAvailable);
+                            DriverTrainingService.waveSource.RecordingStopped += new EventHandler<StoppedEventArgs>(waveSource_RecordingStopped);
+                            DriverTrainingService.waveFile = new WaveFileWriter(createFileName(fileName), waveSource.WaveFormat);
+                            DriverTrainingService.waveSource.StartRecording();                            
+                        }
+                        DriverTrainingService.recordingMetaData.entries.Add(new MetaDataEntry(distanceRoundTrack, recordingName, fileName));
+                        DriverTrainingService.isRecordingSound = true;
+                    }
+                    catch (Exception e)
+                    {
+                        Console.WriteLine("Unable to create a training session sound " + e.Message);
+                    }
+                }
+            }
+        }
+
+        private static String createFileName(String name)
+        {
+            return System.IO.Path.Combine(DriverTrainingService.folderPathForSession, name);
+        }
+
+        static void waveSource_DataAvailable(object sender, WaveInEventArgs e)
+        {
+            lock (_lock)
```

Comment:
you could use this as lock in C# if you like

Reply:
it's from a static context

Reply:
Doh! ;)